docs.feldera.com/docs/pipelines/transactions.md

```diff
@@ -0,0 +1,256 @@
+
+## Transaction API
+
+### Start a transaction
+
+Use the [`start_transaction`](https://docs.feldera.com/api/start-a-transaction) API to start a transaction. The API returns a transaction ID:
+
+<Tabs>
+    <TabItem value="rest" label="REST API">
+    ```shell
+    $ curl -X POST http://localhost:8080/v0/pipelines/my_pipeline/start_transaction
+
+    {"transaction_id":1}
+    ```
+    </TabItem>
+
+    <TabItem value="python" label="Python SDK">
+    ```python
+    from feldera.pipeline import Pipeline
+
+    transaction_id = pipeline.start_transaction()
+    ```
+    </TabItem>
+
+    <TabItem value="fda" label="fda CLI">
+    ```shell
+    $ fda start-transaction my_pipeline
+
+    Transaction started successfully with ID: 1
+    ```
+    </TabItem>
+</Tabs>
+
+* The `start_transaction` call will fail if there is already a transaction in progress.
+
+* During a transaction, the pipeline ingests incoming data without producing output, performing only minimal processing such as resolving primary keys and indexing inputs.
+
+### Commit a transaction
+
+Once the pipeline has ingested all inputs that must be processed as part of the transaction,
+commit the transaction using the [`commit_transaction`](https://docs.feldera.com/api/commit-the-current-transaction) API.
+
+<Tabs>
+    <TabItem value="rest" label="REST API">
+    ```shell
+    $ curl -X POST http://localhost:8080/v0/pipelines/my_pipeline/commit_transaction
+    "Transaction commit initiated"
+    ```
+    </TabItem>
+
+    <TabItem value="python" label="Python SDK">
+    ```python
+    transaction_id = pipeline.commit_transaction(transaction_id, wait = True)
+    ```
+    </TabItem>
+
+    <TabItem value="fda" label="fda CLI">
+    ```shell
+    $ fda commit-transaction my_pipeline
+
+    Transaction committed successfully.
+    ```
+    </TabItem>
+</Tabs>
+
+* The `commit_transaction` REST API initiates the commit. The commit can take some time and complete later.
+  Use the [`/stats` endpoint](#monitoring-transaction-status) to monitor transaction status.
+
+* The Python SDK and `fda` CLI provide both blocking and non-blocking variants of the commit operation.
+  The blocking variant waits until the commit is complete before returning control to the caller.
+
+* During a transaction commit, the pipeline computes updates to all views in the
+  program. Depending on the volume of ingested data and the complexity of the
+  views, this process can take a significant amount of time. While the commit is
+  in progress, the pipeline neither ingests new inputs nor produces outputs. To
+  provide visibility, the pipeline periodically (every 10 seconds) reports the
+  progress of the current commit operation in the log.
```

Comment:
reports -> logs; remove "in the log"?

Mention how the log can be inspected and perhaps give a link